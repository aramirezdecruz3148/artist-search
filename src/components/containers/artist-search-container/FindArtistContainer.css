@import url('https://fonts.googleapis.com/css?family=Monoton|Poiret+One&display=swap');

.findArtist {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.titleh2 {
  font-size: 3vw;
  font-family: 'Monoton', cursive;
  text-align: center;
}

.titleh3 {
  font-size: 3vw;
  font-family: 'Monoton', cursive;
  text-align: center;
}

.findArtist img {
  width: 40vw;
}