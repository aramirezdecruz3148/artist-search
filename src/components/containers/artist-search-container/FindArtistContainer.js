import React, { Component } from 'react';
import { getArtists } from '../../../services/getArtistsDeetsApi';
import SearchArtist from '../../search-artist/SearchArtist';
import Artists from '../../artists/Artists';
import Paging from '../../paging/Paging';
import PropTypes from 'prop-types';

export default class FindArtistContainer extends Component {
  static propTypes = {
    history: PropTypes.object,
    location: PropTypes.object
  }

  state = {
    artistArray: [],
    artist: '',
    loading: false, 
    error: null, 
    page: 1,
    totalPages: 1
  }

  onInputChange = ({ target }) => {
    this.setState({ artist: target.value }, () => {
      this.props.history.push(`/?query=${this.state.artist}`);
    });
  }

  fetchArtists = () => {
    return getArtists(this.state.artist, this.state.page) 
      .then(({ totalPages, singers }) => {
        this.setState({ 
          artistArray: singers, 
          loading: false,
          totalPages: Math.ceil(totalPages / 25)
        });
      })
      .catch(err => this.setState({ 
        error: err, 
        loading: true
      }));
  }

  onButtonClick = () => {
    this.setState({ loading: true, page: 1 });
    return this.fetchArtists();
  }

  changePageCount = (page) => {
    this.setState({ page }); 
    this.props.history.push(`/?query=${this.state.artist}&page=${page}`);
  }

  componentDidMount() {
    const searchArtist = new URLSearchParams(this.props.location.search);
    const queryArtist = searchArtist.get('query');
    const page = parseInt(searchArtist.get('page')) || 1;
    if(queryArtist) {
      this.setState({ artist: queryArtist, page }, () => {
        return this.fetchArtists();
      });
    }
  }

  componentDidUpdate(prevProps, prevState) {
    if(prevState.page !== this.state.page) return this.fetchArtists();
  }

  render() {
    const { 
      artistArray, 
      artist, 
      loading, 
      error,
      page,
      totalPages
    } = this.state;

    if(error) return (
      <>
        <SearchArtist 
          artist={artist} 
          onButtonClick ={this.onButtonClick} 
          onInputChange={this.onInputChange} 
        />
        <h3>Sorry, no artists match that name, why not try another?</h3>
        <img alt='gif of someone listening to music' src='https://media.tenor.com/images/23110dfb65a7f1e3a52a02c41dcc7d2d/tenor.gif'/>
      </>
    );
    if(loading) return <img alt='gif of someone listening to music' src='https://media.tenor.com/images/23110dfb65a7f1e3a52a02c41dcc7d2d/tenor.gif'/>;
    if(totalPages === 1) return (
      <>
        <h2>Find your favorite recording artist...</h2>
        <SearchArtist 
          artist={artist} 
          onButtonClick ={this.onButtonClick} 
          onInputChange={this.onInputChange} 
        />
        <Artists artistArray={artistArray} />
      </>
    );
    if(totalPages === 0) return (
      <>
      <SearchArtist 
        artist={artist} 
        onButtonClick ={this.onButtonClick} 
        onInputChange={this.onInputChange} 
      />
      <h3>Sorry, no artists match that name, why not try another?</h3>
      <img alt='gif of someone listening to music' src='https://media.tenor.com/images/23110dfb65a7f1e3a52a02c41dcc7d2d/tenor.gif'/>
      </>
    );

    return (
      <>
        <h2>Search for your favorite recording artist</h2>
        <SearchArtist 
          artist={artist} 
          onButtonClick ={this.onButtonClick} 
          onInputChange={this.onInputChange} 
        />
        <Paging 
          onClickPrevious={() => this.changePageCount(page - 1)} 
          onClickNext={() => this.changePageCount(page + 1)} 
          disableNext={totalPages === page}
          disablePrev={page === 1}
          currentPage={page}
          totalPages={totalPages}
        />
        <Artists artistArray={artistArray} />
      </>
    );
  }
}
