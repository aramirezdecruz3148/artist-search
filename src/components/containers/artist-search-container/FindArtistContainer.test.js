import React from 'react';
import { shallow } from 'enzyme';
import FindArtistContainer from './FindArtistContainer';

describe('FindArtistContainer component', () => {
  it('renders FindArtistContainer', () => {
    const wrapper = shallow(<FindArtistContainer />);
    expect(wrapper).toMatchSnapshot();
  });
});
