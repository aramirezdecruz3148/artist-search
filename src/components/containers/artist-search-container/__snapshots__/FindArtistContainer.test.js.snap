// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FindArtistContainer component renders FindArtistContainer 1`] = `
<Fragment>
  <SearchArtist
    artist=""
    onButtonClick={[Function]}
    onInputChange={[Function]}
  />
  <Artists
    artistArray={Array []}
  />
</Fragment>
`;
